compiler/pipes/final-check.cpp

```diff
@@ -201,6 +201,34 @@ void check_null_usage_in_binary_operations(VertexAdaptor<meta_op_binary> binary_
       return;
   }
 }
+
+void check_function_test_throws(FunctionPtr f) {
+  std::vector<ClassPtr> throws_expected;
+  for (const auto &class_name : f->test_data->check_throws) {
+    throws_expected.push_back(G->get_class(class_name));
+  }
+  std::vector<ClassPtr> throws_actual;
+  for (auto c : f->exceptions_thrown) {
+    throws_actual.push_back(c);
+  }
+  kphp_error_act(throws_expected.size() == throws_actual.size(), "throws count mismatch", return);
+  auto sort_classes = [](std::vector<ClassPtr> &classes) {
```

Comment:
If I understand correctly, we don't care about the case of catching the base class?

If so, I suggest using `std::unordered_set<ClassPtr>` and his `operator==` for checking equality.